src/controllers/Interactions.ts

```diff
@@ -0,0 +1,103 @@
+  
+  const ballchasers = await QueueRepository.getAllBallChasersInQueue();
+  if (ballchasers == null) {
+
+    await queueChannel.send({ 
+      components: [MessageBuilder.queueButtons],
+      embeds: [MessageBuilder.emptyQueueMessage()] });
+
+  } else {
+
+    await queueChannel.send({ 
+      components: [MessageBuilder.queueButtons],
+      embeds: [MessageBuilder.activeQueueMessage(ballchasers)] });
+  }
+}
+
+NormClient.on("interactionCreate", async (buttonInteraction: Interaction) => {
+
+  if (!buttonInteraction.isButton()) return;
+  await buttonInteraction.deferReply();
+
+  switch (buttonInteraction.customId) {
+    case "joinQueue": {
+
+      const queueMember = await QueueRepository.getBallChaserInQueue(buttonInteraction.user.toString());
+      const leaderboardMember = await LeaderboardRepository.getPlayerStats(buttonInteraction.user.toString());
+
+      if (!queueMember) {
+        const player: BallChaser = {
+          id: buttonInteraction.user.toString(),
+          isCap: false,
+          mmr: leaderboardMember ? leaderboardMember.mmr : 100,
+          name: buttonInteraction.user.username,
+          queueTime: DateTime.now(),
+          team: null
+        };
+        await QueueRepository.addBallChaserToQueue(player);
+      }
+
+      const ballchasers = await QueueRepository.getAllBallChasersInQueue();
+
+      getDiscordChannelById(NormClient, queueChannelId).then((queueChannel) => {
+        if (queueChannel) {
+          queueChannel.messages.delete(buttonInteraction.message.id);
+        }
+      });
+
+      await buttonInteraction.editReply({ 
+        components: [MessageBuilder.queueButtons],
+        embeds: [MessageBuilder.activeQueueMessage(ballchasers)] });
+      break;
+    }
+
+    case "leaveQueue": {
+
+      const member = await QueueRepository.getBallChaserInQueue(buttonInteraction.user.toString());
+
+      if (member != null) {
+        const remainingMembers = await QueueRepository.removeBallChaserFromQueue(buttonInteraction.user.toString());
+
+        getDiscordChannelById(NormClient, queueChannelId).then((queueChannel) => {
+          if (queueChannel) {
+            queueChannel.messages.delete(buttonInteraction.message.id);
+          }
+        });
+
+        await buttonInteraction.editReply({ 
+          components: [MessageBuilder.queueButtons],
+          embeds: [MessageBuilder.activeQueueMessage(remainingMembers)] });
+
+      } else {
+        const ballchasers = await QueueRepository.getAllBallChasersInQueue();
+
+        getDiscordChannelById(NormClient, queueChannelId).then((queueChannel) => {
+          if (queueChannel) {
+            queueChannel.messages.delete(buttonInteraction.message.id);
+          }
+        });
+
+        await buttonInteraction.editReply({
+          components: [MessageBuilder.queueButtons],
+          embeds: [MessageBuilder.activeQueueMessage(ballchasers)] });
+      }
+      break;
+    }
+  }
+});
+
+NormClient.login(process.env.token);
```

Comment:
This is already being run in `index.ts`. Controllers should be agnostic to the client. If you check the LeaderboardChannelController it just accepts the TextChannel to send it to.

I'd recommend just lifting the `NormClient.on("interactionCreate", ...)` to `index.ts` and export a function from `Interactions.ts` to be called on that event. Similar to how the leaderboard controller works.

Reply:
Fixed in next commit